From f576f884943c4e128b3cb49165e7b83ea3990bae Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 20 Apr 2016 15:36:33 +0500
Subject: [PATCH] playbin2/playsink: disable video converters in default config

The video converters misbehave when used through playbin2/playsink
in scenarios where hardware decoding is used. e.g. gstreamer-vaapi
and tend to lockup if playbin is used directly through gst-launch
or from gst-play.
We disable these converters in the default config for playbin2 and
playsink to avoid lockups and provide a better user experience.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 gst/playback/gstplaybin2.c | 2 +-
 gst/playback/gstplaysink.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst/playback/gstplaybin2.c b/gst/playback/gstplaybin2.c
index 2b7416e..28625bf 100644
--- a/gst/playback/gstplaybin2.c
+++ b/gst/playback/gstplaybin2.c
@@ -514,7 +514,7 @@ struct _GstPlayBinClass
 #define DEFAULT_SOURCE            NULL
 #define DEFAULT_FLAGS             GST_PLAY_FLAG_AUDIO | GST_PLAY_FLAG_VIDEO | GST_PLAY_FLAG_TEXT | \
                                   GST_PLAY_FLAG_SOFT_VOLUME | GST_PLAY_FLAG_DEINTERLACE | \
-                                  GST_PLAY_FLAG_SOFT_COLORBALANCE
+                                  GST_PLAY_FLAG_SOFT_COLORBALANCE | GST_PLAY_FLAG_NATIVE_VIDEO
 #define DEFAULT_N_VIDEO           0
 #define DEFAULT_CURRENT_VIDEO     -1
 #define DEFAULT_N_AUDIO           0
diff --git a/gst/playback/gstplaysink.c b/gst/playback/gstplaysink.c
index 37636a9..5431c05 100755
--- a/gst/playback/gstplaysink.c
+++ b/gst/playback/gstplaysink.c
@@ -44,7 +44,7 @@ GST_DEBUG_CATEGORY_STATIC (gst_play_sink_debug);
 #define VOLUME_MAX_DOUBLE 10.0
 
 #define DEFAULT_FLAGS             GST_PLAY_FLAG_AUDIO | GST_PLAY_FLAG_VIDEO | GST_PLAY_FLAG_TEXT | \
-                                  GST_PLAY_FLAG_SOFT_VOLUME | GST_PLAY_FLAG_SOFT_COLORBALANCE
+                                  GST_PLAY_FLAG_SOFT_VOLUME | GST_PLAY_FLAG_SOFT_COLORBALANCE | GST_PLAY_FLAG_NATIVE_VIDEO
 
 #define GST_PLAY_CHAIN(c) ((GstPlayChain *)(c))
 
-- 
1.9.1

